cassandra/cassandra_session.go

```diff
@@ -0,0 +1,137 @@
+package cassandra
+
+import (
+	"sync"
+	"time"
+
+	"github.com/gocql/gocql"
+	log "github.com/sirupsen/logrus"
+)
+
+// Session stores a connection to Cassandra along with associated configurations
+type Session struct {
+	session                 *gocql.Session
+	cluster                 *gocql.ClusterConfig
+	shutdown                chan struct{}
+	connectionCheckTimeout  time.Duration
+	connectionCheckInterval time.Duration
+	addrs                   string
+	logPrefix               string
+	sync.RWMutex
+}
+
+// NewSession creates and returns a CassandraSession
+func NewSession(session *gocql.Session,
+	clusterConfig *gocql.ClusterConfig,
+	shutdown chan struct{},
+	timeout time.Duration,
+	interval time.Duration,
+	addrs string,
+	logPrefix string) *Session {
+	if clusterConfig == nil {
+		panic("NewSession received nil pointer for ClusterConfig")
+	}
+	if session == nil {
+		panic("NewSession received nil pointer for session")
+	}
+
+	cs := &Session{
+		session:                 session,
+		cluster:                 clusterConfig,
+		shutdown:                shutdown,
+		connectionCheckTimeout:  timeout,
+		connectionCheckInterval: interval,
+		addrs:                   addrs,
+		logPrefix:               logPrefix,
+	}
+
+	return cs
+
+}
+
+// DeadConnectionCheck will run a query using the current Cassandra session every connectionCheckInterval
+// if it cannot query Cassandra for longer than connectionCheckTimeout it will create a new session
+//
+// if you are not using a WaitGroup in the caller, just pass in nil
+func (s *Session) DeadConnectionCheck(wg *sync.WaitGroup) {
+	if wg != nil {
+		defer wg.Done()
+	}
+
+	ticker := time.NewTicker(time.Second * s.connectionCheckInterval)
```

Comment:
`s.connectionCheckInterval` is a `time.Duration`, so I don't think this needs to be multiplied with any unit.

Reply:
Ooops thanks, I think I missed that when I changed it to be a Duration when I was copying it around.